src/services/rest/RestService.ts

```diff
@@ -0,0 +1,148 @@
+
+  constructor(private $http: ng.IHttpService,
+              private $cacheFactory: ng.ICacheFactoryService,
+              private $q: ng.IQService,
+              private $location: ng.ILocationService,
+              private socketBindingService: SocketBindingService) {
+
+    // TODO somehow discover installed rest binding better than this?
+    socketBindingService.getDefaultSocketBindingGroup().then((result: ISocketBinding []) => {
+      angular.forEach(result, (value) => {
+        if (value.name === "rest") {
+          this.restBinding = value;
+        }
+      });
+    });
+  }
+
+  executeCacheGet(cacheTarget: string, key: string): ng.IPromise<any> {
+    let request: IRestRequest = {
+      jsonPayload: undefined,
+      cache: cacheTarget,
+      urlSuffix: "/".concat(key)
+    };
+    let url: string = this.generateUrl(request);
+    return this.executeHelper("get", url, request.jsonPayload, this.getDefaultHeaders());
+  }
+
+  executeCacheDelete(cacheTarget: string): ng.IPromise<any> {
+    let request: IRestRequest = {
+      jsonPayload: undefined,
+      cache: cacheTarget,
+      urlSuffix: undefined
+    };
+    let url: string = this.generateUrl(request);
+    return this.executeHelper("delete", url, request.jsonPayload, this.getDefaultHeaders());
+  }
+
+  executeCacheDeleteKey(cacheTarget: string, key: string): ng.IPromise<any> {
+    let request: IRestRequest = {
+      jsonPayload: undefined,
+      cache: cacheTarget,
+      urlSuffix: "/".concat(key)
+    };
+    let url: string = this.generateUrl(request);
+    return this.executeHelper("delete", url, request.jsonPayload, this.getDefaultHeaders());
+  }
+
+  executeCachePut(cacheTarget: string, key: string, data: any): ng.IPromise<any> {
+    let request: IRestRequest = {
+      jsonPayload: data,
+      cache: cacheTarget,
+      urlSuffix: "/".concat(key)
+    };
+    let url: string = this.generateUrl(request);
+    return this.executeHelper("put", url, request.jsonPayload, this.getDefaultHeaders());
+  }
+
+  executeCachePost(cacheTarget: string, key: string, data: any): ng.IPromise<any> {
+    let request: IRestRequest = {
+      jsonPayload: data,
+      cache: cacheTarget,
+      urlSuffix: "/".concat(key)
+    };
+    let url: string = this.generateUrl(request);
+    return this.executeHelper("post", url, request.jsonPayload, this.getDefaultHeaders());
+  }
+
+  executeCacheQuery(cacheTarget: string, query: string): ng.IPromise<any> {
+    let request: IRestRequest = {
+      jsonPayload: {query: query},
+      cache: cacheTarget,
+      urlSuffix: "?action=search"
+    };
+    let url: string = this.generateUrl(request);
+    return this.executeHelper("post", url, request.jsonPayload, this.getDefaultHeaders());
+  }
+
+  clearGetCache(): void {
+    this.$cacheFactory.get("$http").removeAll();
+  }
+
+  private executeHelper(type: string, url: string, data: any, headers?: ng.IRequestShortcutConfig): ng.IPromise<any> {
```

Comment:
It seems that `getDefaultHeaders` is always passed to `executeHelper`, so we might as well just define the headers variable here instead of passing as a param.

Reply:
This is still the case.

Reply:
@vblagoje I think you edited my comment :smile: 

What I mean is that headers is always equal to `getDefaultHeaders`, so you might as well just define headers in the method once opposed to passing it everytime, i.e.:
```typescript
private executeHelper(type: string, url: string, data: any): ng.IPromise<any> {
let headers: ng.IRequestShortcutConfig = {
    headers: {
        "Accept": "*/*",
        "Content-type": "application/json"
    }
};
...
}
```